### DISCUSSION TITLE
Determine if auto-renew status or not

### DISCUSSION THREAD

**Discussion by lahivee** [imdhemy/laravel-in-app-purchases#32] [Q&A] [answered]
Hi,

when receiving the DID_CHANGE_RENEWAL_STATUS event, how do you know if it is auto-renewing or not?
(for both Apple and Google?)

I did not find any methods or properties on the PurchaseEventContract.

Thank you

**Comment by imdhemy** [comment, marked as answer]
Hi @lahivee ,

Firstly,
The event names are named after the notification types named by each provider.
Google does not send a `DID_CHANGE_RENEWAL_STATUS` notification, but Apple does, as shown in the configuration section.

Kindly, check the following links to find the correct event to handle:
- [For Google](https://developer.android.com/google/play/billing/rtdn-reference#sub)
- [For Apple](https://developer.apple.com/documentation/appstoreservernotifications/notification_type)

## To handle the subscription auto-renewal:

**For Google**, I suggest using [SUBSCRIPTION_RENEWED](https://github.com/imdhemy/laravel-in-app-purchases/blob/ed03a4a128aa36b5b46287b235d3bbf77545fe63/config/purchase.php#L41):

```
'eventListeners' => [
    SubscriptionRenewed::class => [YourAwesomeListener::class],
],
```

**For Apple**, I suggest using [DID_RENEW](https://github.com/imdhemy/laravel-in-app-purchases/blob/master/config/purchase.php#L64) and [INTERACTIVE_RENEWAL](https://github.com/imdhemy/laravel-in-app-purchases/blob/master/config/purchase.php#L66). _Check the provided links for more information._

```
'eventListeners' => [
    InteractiveRenewal::class => [YourAwesomeListener::class],
    DidRenew::class => [YourAwesomeListener::class],
],
```

## DID_CHANGE_RENEWAL_STATUS
Apple says,

> DID_CHANGE_RENEWAL_STATUS
Indicates a change in the subscription renewal status. In the JSON response, check auto_renew_status_change_date_ms to know the date and time of the last status update. Check auto_renew_status to know the current renewal status

Make sure you are using `imdhemy/laravel-purchases ` of version: `0.8.6`:

Inside your event listener `handle()` method:

```php
class DidChangeRenewalStatusListener
{
    /**
     * @param DidChangeRenewalStatus $event
     */
    public function handle(DidChangeRenewalStatus $event): void
    {
        $isAutoRenewal = $event->isAutoRenewal();
        $changeDate = $event->getAutoRenewStatusChangeDate();
        
        // ...
    }
}
```

**Reply by lahivee** [reply]
Thank you for the update @imdhemy

I think this addition in 0.8.6 was important because Apple docs state that DID_CHANGE_RENEWAL_STATUS is also used to cancel a subscription:

"User disabled subscription auto-renewal in the App Store account’s Subscriptions settings, effectively canceling the subscription"
(Source: https://developer.apple.com/documentation/appstoreservernotifications/notification_type)

I will experiment with this as soon as I can and really appreciate you for this package and your maintainer efforts!
